File: BMI_v3/app/src/main/java/com/example/bmi_v3/Bmi.java
package com.example.bmi_v3;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class Bmi extends AppCompatActivity {
    EditText weight, height;
    TextView resulttext;
    String calculation, BMIresult;
    Button calculate_button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmi);

        weight = findViewById(R.id.weight);
        height = findViewById(R.id.height);
        resulttext = findViewById(R.id.result);
        calculate_button = findViewById(R.id.calculate_button);

        calculate_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculateBMI();
            }
        });


        //initialize and assign variable
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);

        //set Home Selected
        bottomNavigationView.setSelectedItemId(R.id.bmi);

        //perform itemselectedlistener
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                switch (menuItem.getItemId()){
                    case R.id.bmi:
                        return true;
                    case R.id.home:
                        startActivity(new Intent(getApplicationContext(),MainActivity.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.bmr:
                        startActivity(new Intent(getApplicationContext(), Bmr.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.recipes:
                        startActivity(new Intent(getApplicationContext(), Recipes.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.quiz:
                        startActivity(new Intent(getApplicationContext(), Quiz.class));
                        overridePendingTransition(0,0);
                        return true;

                }
                return false;
            }
        });



    }
    public void calculateBMI() {
        String S1 = weight.getText().toString();
        String S2 = height.getText().toString();

        float weightValue = Float.parseFloat(S1);
        float heightValue = Float.parseFloat(S2);

        float bmi = weightValue / (heightValue/100 * heightValue/100);

        if (bmi < 18.5){
            BMIresult = "Under Weight";
        }
        else if (bmi >= 18.5 && bmi <= 24.9) {
            BMIresult = "Your Weight is fine";
        }
        else if (bmi >=25 && bmi <=29.9){
            BMIresult = "Over Weight";
        }
        else {
            BMIresult = "U are really FAT";
        }

        calculation = "Result: \n\n" + String.format("%.2f", bmi) + "\n" + BMIresult;
        resulttext.setText(calculation);
    }
}

File: kotlin/BMI_v3/app/src/main/java/com/example/bmi_v3/Bmi.kt
package com.example.bmi_v3

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_bmi.*

class Bmi : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmi)

        calculate_button.setOnClickListener { calculateBMI() }

        //initialize and assign variable
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        //set Home Selected
        bottomNavigationView.selectedItemId = R.id.bmi
        //perform itemselectedlistener
        bottomNavigationView.setOnNavigationItemSelectedListener(BottomNavigationView.OnNavigationItemSelectedListener { menuItem ->
            when (menuItem.itemId) {
                R.id.bmi -> return@OnNavigationItemSelectedListener true
                R.id.home -> {
                    startActivity(Intent(applicationContext, MainActivity::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
                R.id.bmr -> {
                    startActivity(Intent(applicationContext, Bmr::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
                R.id.recipes -> {
                    startActivity(Intent(applicationContext, Recipes::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
                R.id.quiz -> {
                    startActivity(Intent(applicationContext, Quiz::class.java))
                    overridePendingTransition(0, 0)
                    return@OnNavigationItemSelectedListener true
                }
            }
            false
        })
    }

    private fun calculateBMI() {
        val s1 = weight.text.toString()
        val s2 = height.text.toString()
        val bmiResult:String
        val calculation:String

        val weightValue = s1.toFloat()
        val heightValue = s2.toFloat()

        val bmi = weightValue / (heightValue / 100 * heightValue / 100)

        when (bmi.toInt()){
            in 0..19 -> bmiResult = "Under Weight"
            in 19..25 -> bmiResult = "Your Weight is fine"
            in 25..30 -> bmiResult = "Over Weight"
            else -> bmiResult = "U are really FAT"}
            calculation = "Result: \n\n" + String.format("%.2f", bmi) + "\n" + bmiResult
            result.text = calculation
    }
}